### DISCUSSION TITLE
DIO2  as RF switch behavior (sx126x)

### DISCUSSION THREAD

**Discussion by devRomeos** [jgromes/RadioLib#1312] [General]
I am trying the [RA-01SH](https://docs.ai-thinker.com/_media/lora/docs/ra-01sh_specification.pdf) modules for the first time and although everything seems to be working fine, I don't think I am getting the best signal / sensitivity when transmitting and receiving.

From the datasheet I can confirm DIO2 can manage the RF control so I used _setDio2AsRfSwitch(true)_ command however, I'm getting the same signal levels with _setDio2AsRfSwitch(false)_ or omitting this command altogether, which makes me somewhat suspect...

![image](https://github.com/user-attachments/assets/9d72f8a6-2188-45eb-8741-1ad72bcb4e1f)

I've measured DIO2 with scope and I don't see any signal change when transmitting or receiving, it always stays at 0V. 

Can someone tell me what I should expect in DIO2 or also how I can confirm that the RF switch is working?
Thanks

**Comment by HeadBoffin** [comment]
What MCU are you using and do you have DIO2 wired up?

**Reply by devRomeos** [reply]
Esp32, DIO2 is not wired up as it is supposed to be handled by the SX itself in this mode.

I realize the HIGH on DIO2 was so fast that I didn't see it in the scope, so if it seems to be working correctly, I'm marking this as solved.
